### PR TITLE
[Backport 3.6] Move some scripts to framework submodule

### PR DESCRIPTION
**Pull request by davidhorstmann-arm**
Trivial backport of #8991

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required - internal changes only
- [x] **3.6 backport** of #8991 
- [x] **2.28 backport** not required - no framework submodule in 2.28
- [x] **tests** not required - file moves only

### CODE REVIEW

**Review by Ryan-Everett-arm — APPROVED** [review]
LGTM

**Review by Ryan-Everett-arm — CHANGES_REQUESTED** [review]
Looks like there are a couple of CI failures with file generation. I think the cause is this:
 There is a file `tests/scripts/generate_psa_wrappers.py` which exists in this branch ([and development](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/scripts/generate_psa_wrappers.py)) but not in the branch for #8991, hence the CI passing there. This file has references to `mbedtls_dev` which need to be changed to `framework_dev`. I'm not sure why the file doesn't exist in #8991 - I think that #8991 also needs to be rebased so that we can change the `mbedtls_dev` references there.

File: framework
Comment (Ryan-Everett-arm) [inline framework:1]: `c_parsing_helper.py` and `c_wrapper_generator.py` are added to framework but are not currrently removed from `mbedtls_dev`

**Review by Ryan-Everett-arm — CHANGES_REQUESTED** [review]
Thanks for the changes. It seems that there are now two scripts in both `mbedtls_dev` and `framework_dev`, everything else looks good though!

**Review by Ryan-Everett-arm — APPROVED** [review]
LGTM, thanks for the quick changes!

**Comment by ronald-cron-arm** [comment]
Closing as superseded by more separate PRs

**Pull request closed without merging**